docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+    - This method will retrieve the flags entered by User, at the same time validating the User input. The criteria 
+    entered by the User will be returned as flags, an Array of Strings of fixed size 5 as the software only supports 5 
+    flags in version 2.
+- isAvailStudyArea() : 
+    - This method does the check mentioned above and returns a boolean value. Based on the flags detected, 
+    the method will iterate through the flags array and check if the Study Area's attribute matches each flag stated in
+    flags. If isAvailStudyArea returns as true for all flags, the Study Area is then added to the output list, 
+    availStudyArea. 
+
+Below would be a sequence diagram to demonstrate how the search algorithm is operated.
+
+ * `User enters search key` 
+ 
+![Study Area Sequence_Diagram_Main](images/user_sL_interaction.png)  
+<div>Figure 9. Interaction between User and Study Area Search Interface</div>
+<br>
+
+ * `StudyAreaCommand invokes searchList() of StudyAreaList` 
+ 
+![Study_Area_Sequence_Diagram_subModules](images/sL_sA_interaction.png)
+<div>Figure 10. Interaction within Study Area Search Interface</div>
+<br>
+
+![Study_Area_Sequence_Diagram_subModules2](images/isAvailStudyArea.png)
+<div>Figure 11. Interaction when isAvailStudyArea is invoked</div>
+<br>
+You can refer [here](#appendix-d-glossary) for a detailed explanation on the terms used in this diagram
+#### 3.2.2 Alternative 
+Aspect: How to search based on User input.
+
+- Alternative 1 (Current Choice) :
+ 
+    Iterate through the list of all existing Study Areas. Then check if each Study Area 
+meets User's Criteria
+    - Pros : Lesser data structures required. Therefore lesser memory required.
+    - Cons : Linear search, therefore, with bigger size of data, search may take longer. 
+    
+- Alternative 2 : 
+    
+    Create adjacency lists based on Study Area attributes. If the Study Area contains that attribute, the 
+Study Area is added in that specific attribute list. The creation and initialisation of attribute lists are done when
+the software is setting up. For example : Locations -> Study Area. Therefore, each attribute has its own list. Based on
+user criteria, concatenate the output list with the Study Areas in the related attribute list. If more than one 
+requirement is entered by the user, only the Study Areas appears in all the related attributes specified by the User 
+will be added to the output list.
+    - Pros : Data is categorised base on attributes
+    - Cons : More memory is required as more data structures are used. Since methods invoked during the search are also 
+    linear, time taken to complete the search will be longer if more data are present. Overall, this approach is harder 
+    to implement.
+  
+Therefore, the first alternative is chosen, as it is easier to implement and lesser memory is used while 
+conducting the search.
+
+[comment]: # (@@author terrytay)
+
+### 3.3. Operation of Notes
+
+#### 3.3.1 Implementation 
```

Comment:
This sections was a little too wordy, it might be better to have simple diagrams to help envision the flow as with the diagrams at the start.